specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -0,0 +1,5214 @@
+    "ModelEnvironmentDefinition": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the environment.",
+          "type": "string",
+          "example": "mydevenvironment"
+        },
+        "version": {
+          "description": "The environment version.",
+          "type": "string",
+          "example": "1"
+        },
+        "python": {
+          "description": "Settings for a Python environment.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/ModelPythonSection"
+            }
+          ]
+        },
+        "environmentVariables": {
+          "description": "Definition of environment variables to be defined in the environment.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "docker": {
+          "description": "The definition of a Docker container.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/ModelDockerSection"
+            }
+          ]
+        },
+        "spark": {
+          "description": "The configuration for a Spark environment.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/ModelSparkSection"
+            }
+          ]
+        },
+        "r": {
+          "description": "Settings for a R environment.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/RSection"
+            }
+          ]
+        },
+        "inferencingStackVersion": {
+          "description": "The inferencing stack version added to the image. To avoid adding an inferencing stack, do not set this value. Valid values: \"latest\".",
+          "type": "string",
+          "example": "latest"
+        }
+      }
+    },
+    "EnvironmentReference": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the environment.",
+          "type": "string"
+        },
+        "version": {
+          "description": "Version of the environment.",
+          "type": "string"
+        }
+      }
+    },
+    "ModelPythonSection": {
+      "type": "object",
+      "properties": {
+        "interpreterPath": {
+          "description": "The python interpreter path to use if an environment build is not required. The path specified gets used to call the user script.",
+          "type": "string"
+        },
+        "userManagedDependencies": {
+          "description": "True means that AzureML reuses an existing python environment; False means that AzureML will create a python environment based on the Conda dependencies specification.",
+          "type": "boolean"
+        },
+        "condaDependencies": {},
+        "baseCondaEnvironment": {
+          "type": "string"
+        }
+      }
+    },
+    "ContainerRegistry": {
+      "type": "object",
+      "properties": {
+        "address": {
+          "type": "string"
+        },
+        "username": {
```

Comment:
password and maybe username should NOT be returned on PUT/GET responses

Reply:
Previously request and response shared the same definition. Made changes to use separate definitions. Response no longer has these properties.